fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/model/ProcedureDef.java

```diff
@@ -44,17 +40,23 @@ public void apply(IDatabaseAdapter target) {
         }
     }
 
-    /* (non-Javadoc)
-     * @see com.ibm.fhir.database.utils.model.IDatabaseObject#drop(com.ibm.fhir.database.utils.api.IDatabaseAdapter)
-     */
+    @Override
+    public void apply(Integer priorVersion, IDatabaseAdapter target) {
+        // Serialize the execution of the procedure, to try and avoid the
+        // horrible deadlocks we keep getting
+        synchronized(target) {
+            // TODO migration logic here?
```

Comment:
Yes, pre-migration logic makes sense here. 
As does post-migration logic.

Reply:
pre and post-migration logic has been replaced by a single list of "migrations" for now.
if we end up needing pre and post-migration logic for procedures, we'll need to revisit that later.

for now, i'm just logging a warning if we get to `ProcedureDef.apply(priorVersion, target)` with migration steps, because instead it just calls apply (which invokes `create or replace`)